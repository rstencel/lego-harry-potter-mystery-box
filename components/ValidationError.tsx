interface ValidationErrorProps {
  error?: string
}

const ValidationError = ({ error }: ValidationErrorProps) => (
  <p className="error" style={{ visibility: error ? "visible" : "hidden"}}>
    {error}
  </p>
)

export default ValidationError