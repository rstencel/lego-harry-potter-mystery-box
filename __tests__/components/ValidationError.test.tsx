import { render } from "@testing-library/react"
import ValidationError from "../../components/ValidationError"

describe("ValidationError", () => {
  const mockError = "error"

  describe("Should match snapshot", () => {
    it("- hidden", () => {
      const { container } = render(<ValidationError error={undefined} />)
      expect(container).toMatchSnapshot()
    })
    
    it("- visible", () => {
      const { container } = render(<ValidationError error={mockError} />)
      expect(container).toMatchSnapshot()
    })
  })

  it("Should display error", () => {
    const { container } = render(<ValidationError error={mockError} />)
    expect(container).toHaveTextContent(mockError)
  })
})
