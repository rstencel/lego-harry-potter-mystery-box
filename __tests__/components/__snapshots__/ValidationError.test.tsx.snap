// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationError Should match snapshot - hidden 1`] = `
<div>
  <p
    class="error"
    style="visibility: hidden;"
  />
</div>
`;

exports[`ValidationError Should match snapshot - visible 1`] = `
<div>
  <p
    class="error"
    style="visibility: visible;"
  >
    error
  </p>
</div>
`;
